### PR TITLE
[IMP] mail: Allow to easily override message post data

### PR DESCRIPTION
**Pull request by jat-odoo**
Description of the issue/feature this PR addresses:

Current behavior before PR:
It is not possible to pass custom values in message post from composer.

Desired behavior after PR is merged:
It will allow to easily override and pass custom values in message post.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by jat-odoo** [comment]
@alexkuhn Can you check it please?

File: addons/mail/controllers/discuss.py [hunk 1]
```diff
@@ -207,15 +207,17 @@ def discuss_starred_messages(self, max_id=None, min_id=None, limit=30, **kwargs)
     # Thread API (channel/chatter common)
     # --------------------------------------------------------------------------
 
+    def _get_alllowed_message_post_params(self):
```
Comment (alexkuhn) [inline addons/mail/controllers/discuss.py:210]: `alllowed` => `allowed`

File: addons/mail/static/src/models/composer_view/composer_view.js [hunk 2]
```diff
@@ -364,6 +348,29 @@ function factory(dependencies) {
             }
         }
 
+        /**
+         * Gather data for message post.
+         */
+         _getPostData(composer) {
```
Comment (alexkuhn) [inline addons/mail/static/src/models/composer_view/composer_view.js:354]: Don't need passing `composer`, you can just use `this.composer`

Comment (alexkuhn) [inline addons/mail/static/src/models/composer_view/composer_view.js:354]: Missing `@private`

Comment (alexkuhn) [inline addons/mail/static/src/models/composer_view/composer_view.js:354]: Should be placed alphabetically in the file, in the proper `private` category

Comment (alexkuhn) [inline addons/mail/static/src/models/composer_view/composer_view.js:354]: `_getPostData` => `_getMessageData`

**Comment by alexkuhn** [comment]
Why do you need to override data in message post?

**Comment by jat-odoo** [comment]
@alexkuhn I'm working on project and now i have case to pass custom values in message post.
So, Made it overridable that also help someone in future.

**Review by alexkuhn — CHANGES_REQUESTED** [review]
Such changes are usually better in master. Is it good for you doing it in master?

**Comment by alexkuhn** [comment]
Could you please add a commit to sign CLA?
See https://github.com/odoo/odoo/blob/master/doc/cla/sign-cla.md
Thanks!

**Comment by jat-odoo** [comment]
done!

**Comment by mart-e** [comment]
@jat-odoo would you mind changing your github handle? The `-odoo` suffix leads to some confusions, as you have seen with Alexandre.
Good luck in your new projects :)

**Comment by jat-odoo** [comment]
Hi @mart-e ,

I created this `jat-odoo` Github user 8 years ago. this is only my Github account. I have many repos/projects/tools/scripts running on this username. So, I don't want to break anything.

I hope, you understand.

Thanks.

**Review by alexkuhn — APPROVED** [review]
@robodoo r+ rebase-ff

**Pull request closed without merging**